src/lib/signal-state.ts

```diff
@@ -16,6 +17,7 @@ export type SignalStateUpdate<State extends Record<string, unknown>> = {
 export function signalState<State extends Record<string, unknown>>(
   initialState: State
 ): SignalState<State> {
+  deepFreeze(initialState);
```

Comment:
we should perform `deepFreeze` only in dev mode for better performance

Reply:
Still working on that